src/sage/graphs/graph_decompositions/slice_decomposition.pyx

```diff
@@ -0,0 +1,970 @@
+        [0[1[4[5]]][2[6]][3][9][7][8]]
+
+    The graph can have loops or multiple edges but they are ignored::
+
+        sage: H = Graph(G,loops=True,multiedges=True)
+        sage: H.add_edges([(4, 4), (2, 2), (1, 6)])
+        sage: SD2 = H.slice_decomposition()
+        sage: SD2 == SD
+        True
+        sage: SD2.underlying_graph() == G.to_simple(immutable=True)
+        True
+
+    The tree corresponding to the slice decomposition can be displayed using
+    ``view``::
+
+        sage: from sage.graphs.graph_latex import check_tkz_graph
+        sage: check_tkz_graph() # random - depends on Tex installation
+        sage: view(G) # not tested
+        sage: latex(G) # to obtain the corresponding LaTeX code
+        \begin{tikzpicture}
+        ...
+        \end{tikzpicture}
+
+    Slice decompositions are only defined for undirected graphs::
+
+        sage: from sage.graphs.graph_decompositions.slice_decomposition import slice_decomposition
+        sage: slice_decomposition(DiGraph())
+        Traceback (most recent call last):
+        ...
+        ValueError: Graph must be undirected
+
+    """
+    return SliceDecomposition(G, initial_vertex=initial_vertex)
+
+
+cdef class SliceDecomposition(SageObject):
+
+    def __init__(self, G, initial_vertex=None):
+        r"""
+        Represents a slice decomposition of a simple directed graph.
+
+        INPUT:
+
+        - ``G`` -- a Sage graph.
+
+        - ``initial_vertex`` -- (default: ``None``); the first vertex to
+          consider.
+
+        See :meth:`~slice_decomposition` for more details.
+
+        .. automethod:: __getitem__
+        """
+        if G.is_directed():
+            raise ValueError("Graph must be undirected")
+
+        if initial_vertex is not None and initial_vertex not in G:
+            raise LookupError(f"vertex ({initial_vertex}) is not a vertex of the graph")
+
+        cdef CGraphBackend Gbackend = <CGraphBackend> G._backend
+        cdef CGraph cg = Gbackend.cg()
+
+        self._graph_class = type(G)
+
+        cdef int initial_v_int
+        if initial_vertex is not None:
+            # we already checked that initial_vertex is in G
+            initial_v_int = Gbackend.get_vertex(initial_vertex)
+        else:
+            initial_v_int = -1
+
+        cdef vector[int] sigma
+        cdef vector[vector[int]] lex_label
+
+        # Compute the slice decomposition using the extended lexBFS algorithm
+        extended_lex_BFS(cg, sigma, NULL, initial_v_int, NULL,
+                         &(self.xslice_len), &lex_label)
+
+        # Translate the results with the actual vertices of the graph
+        self.sigma = tuple(Gbackend.vertex_label(v_int) for v_int in sigma)
+        self.sigma_inv = {v: i  for i, v in enumerate(self.sigma)}
+        self.lex_label = {i: tuple(Gbackend.vertex_label(v_int) for v_int in lli)
+                                    for i, lli in enumerate(lex_label)}
+
+    def __eq__(self, other):
+        """
+        TESTS:
```

Comment:
Add 1 line description of the method.

Reply:
done in 306c3254289d0d3240ae05049ad653fb7756dbfa